lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -719,30 +721,24 @@ void printSubstraitType(AsmPrinter &printer, Operation *op,
   });
 }
 
-} // namespace
-
 //===----------------------------------------------------------------------===//
 // Substrait operations
 //===----------------------------------------------------------------------===//
 
 #define GET_OP_CLASSES
 #include "substrait-mlir/Dialect/Substrait/IR/SubstraitOps.cpp.inc" // IWYU pragma: keep
 
-namespace {
-
 /// Computes the type of the nested field of the given `type` identified by
 /// `position`. Each entry `n` in the given index array `position` corresponds
 /// to the `n`-th entry in that level.
-FailureOr<Type> computeTypeAtPosition(Location loc, Type type,
-                                      ArrayRef<int64_t> position);
-
-namespace impl {
```

Comment:
We should keep this to make clear that the functions in this namespace are an implementation detail.